src/Jackalope/Version/VersionHandler.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Jackalope\Version;
+
+use Jackalope\Node;
+use Jackalope\Transport\TransportInterface;
+use PHPCR\UnsupportedRepositoryOperationException;
+
+class VersionHandler
+{
+    const MIX_VERSIONABLE = 'mix:versionable';
+    const MIX_SIMPLE_VERSIONABLE = 'mix:simpleVersionable';
+
+    /**
+     * @var TransportInterface
+     */
+    private $transport;
+
+    public function __construct(TransportInterface $transport)
+    {
+        $this->transport = $transport;
+    }
+
+    public static function addVersionProperties(Node $node, TransportInterface $transport)
+    {
+        // TODO solve without reflection
+        $reflection = new \ReflectionMethod(get_class($transport), 'storeNode');
+        $reflection->setAccessible(true);
+
+        // TODO move this to the version handler
+        $session = $node->getSession();
+        $rootNode = $session->getRootNode();
+
+        $node->setProperty('jcr:isCheckedOut', true);
+
+        if (!$rootNode->hasNode('jcr:system/jcr:versionStorage')) {
+            if (!$rootNode->hasNode('jcr:system')) {
+                $rootNode->addNode('jcr:system');
+            }
+            $rootNode->addNode('jcr:system/jcr:versionStorage');
```

Comment:
If we did, it could be when the (workspace is initiated)[https://github.com/jackalope/jackalope-doctrine-dbal/blob/master/src/Jackalope/Transport/DoctrineDBAL/Client.php].